src/android/BluetoothLePlugin.java

```diff
@@ -1478,11 +1478,11 @@ private void connectAction(JSONArray args, CallbackContext callbackContext) {
       autoConnect = obj.optBoolean("autoConnect", false);
     }
 
+    connections.put(device.getAddress(), connection);
+
```

Comment:
So sometimes the callback is called is called immediately, and thus the connection doesn't exist in the map yet?

Reply:
Yes, sometimes the callback is called immediately,this the connection doesn't exist in the map yet. 
(newState at this time is BluetoothProfile.STATE_CONNECTED, but connection is null )

- BluetoothLePlugin.java
```
  private BluetoothGattCallback bluetoothGattCallback = new BluetoothGattCallback() {
    @Override
    public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState) {
       ...
      HashMap<Object, Object> connection = connections.get(address);
      if (connection == null) {
        return;
      }
      ...
```

Reply:
Thank you, great catch!

Reply:
Thanks for merging.

Reply:
Thank you very much for release！